src/devices/cpu/e132xs/e132xsop.hxx

```diff
@@ -3241,167 +1571,328 @@ void hyperstone_device::hyperstone_andni_global_simm()
 	m_icount -= m_clock_cycles_1;
 }
 
-void hyperstone_device::hyperstone_andni_global_limm()
+template <hyperstone_device::reg_bank DST_GLOBAL, hyperstone_device::reg_bank SRC_GLOBAL>
+void hyperstone_device::hyperstone_neg()
 {
-	const uint32_t extra_u = decode_immediate_s();
-
 	check_delay_PC();
 
-	uint32_t imm;
-	if (N_OP_MASK == 0x10f)
-		imm = 0x7fffffff; // bit 31 = 0, others = 1
-	else
-		imm = extra_u;
+	const uint32_t fp = GET_FP;
+	const uint32_t src_code = SRC_GLOBAL ? SRC_CODE : ((SRC_CODE + fp) & 0x3f);
+	const uint32_t dst_code = DST_GLOBAL ? DST_CODE : ((DST_CODE + fp) & 0x3f);
 
-	const uint32_t dst_code = DST_CODE;
-	const uint32_t dreg = m_global_regs[dst_code] & ~imm;
-	set_global_register(dst_code, dreg);
+	const uint32_t sreg = SRC_GLOBAL ? ((src_code == SR_REGISTER) ? GET_C : m_global_regs[src_code]) : m_local_regs[src_code];
+	const uint64_t tmp = -uint64_t(sreg);
+
+	SR &= ~(C_MASK | V_MASK | Z_MASK | N_MASK);
+	SR |= (tmp & 0x100000000L) >> 32;
+	SR |= (tmp & sreg & 0x80000000) >> 28;
+
+	const uint32_t dreg = -sreg;
+	(DST_GLOBAL ? m_global_regs : m_local_regs)[dst_code] = dreg;
```

Comment:
This opcode handler assigns into m_global_regs directly rather than using set_global_register - is that intentional or an oversight?